File: adminportal/src/Dto/AdminPortal/RequestInfo.ts
import type { TriageStatus, TriageType } from "./UpdateRequestTriageStatusRequest";
import type { UserBasicInfo, UserEntry } from "./UserInfo";

export type RequestEntry = {
    requestId: string;
    createdDate: Date;
    supportRequestTitle: string;
    supportRequestContent: string;
    dateUpdated: Date;
    dateResolved: Date;
    isMarkedForDelete: boolean;
    dateMarkedForDelete: Date;
    triageStatus: TriageStatus;
    triageType: TriageType;
    userRequestMappings: UserRequestMappingEntry[];
    createdByUser: UserEntry;
    assignedToUser: UserEntry;
    resolvedByUser: UserEntry;
}

export type UserRequestMappingEntry = {
    UserMappingId: string,
    user: UserEntry,
    userId: string,
    request: RequestEntry,
    requestId: string,
    isCreator: boolean,
    isAssigned: boolean,
    isResolved: boolean,
    isClosed: boolean,
    dateAssigned: Date
}

export type RequestBasicInfo = {
    request: RequestEntry,
    createdByUser: UserBasicInfo,
}

export type RequestSummary = {
    totalRequests: number,
    numResolvedRequests: number,
    numAssignedRequests: number,
    numPendingRequests: number,   
}

File: server/RestApiServer.Endpoints/Dto/Admin/RequestInfo.cs
using RestApiServer.Db;
using RestApiServer.Db.Users;

namespace RestApiServer.Dto.Admin
{
    public class RequestBasicInfo
    {
        public required RequestEntry Request { get; set; }
    }
    public class RequestFullInfo
    {
        public required RequestEntry Request { get; set; }
        public UserEntry CreatedByUser { get; set; } = new();
        public UserEntry AssignedToUser { get; set; } = new();
        public UserEntry ResolvedByUser { get; set; } = new();
        public UserEntry LastUpdatedByUser { get; set; } = new();        
    }

    public class RequestSummary
    {
        public int TotalRequests { get; set; } = 0;
        public int NumResolvedRequests { get; set; } = 0;
        public int NumAssignedRequests { get; set; } = 0;
        public int NumPendingRequests { get; set; } = 0;
    }
}